src/shogun/mathematics/linalgrefactor/CPUBackend.cpp

```diff
@@ -0,0 +1,16 @@
+#include <shogun/mathematics/linalgrefactor/CPUBackend.h>
+
+namespace shogun
+{
+
+template <typename T> T CPUBackend::dot(const CPUVector<T> &a, const CPUVector<T> &b)
```

Comment:
Usually in shogun the style is to put a newline after the `template <typename T>`. Make similar changes in other places where you've used this